src/lib/data-collector.js

```diff
@@ -43,10 +43,14 @@ DataCollector.prototype.log = function (message) {
 
 DataCollector.prototype.getDeviceData = function () {
   if (!this._instance) {
-    return '';
+    return Promise.resolve('');
   }
 
-  return this._instance.deviceData;
+  // TODO do we want to maintain the behavior
+  // where if data collector instantiation fails
+  // we just return an empty string for for device data
+  // or should we actually error now?
+  return this._instance.getDeviceData();
```

Comment:
If the original code had undefined behavior, we may be able to replace this TODO with a JIRA ticket and define some new acceptance criteria.